File: app/src/main/java/com/androidacademy/msk/exerciseproject/screen/news_list/NewsListView.java
package com.androidacademy.msk.exerciseproject.screen.news_list;

import android.support.annotation.NonNull;

import com.androidacademy.msk.exerciseproject.data.database.entity.DbNewsItem;
import com.arellomobile.mvp.MvpView;
import com.arellomobile.mvp.viewstate.strategy.AddToEndStrategy;
import com.arellomobile.mvp.viewstate.strategy.OneExecutionStateStrategy;
import com.arellomobile.mvp.viewstate.strategy.SingleStateStrategy;
import com.arellomobile.mvp.viewstate.strategy.StateStrategyType;

import java.util.List;

public interface NewsListView extends MvpView {

    @StateStrategyType(SingleStateStrategy.class)
    void showProgressBar();

    @StateStrategyType(SingleStateStrategy.class)
    void showNews(@NonNull List<DbNewsItem> news);

    @StateStrategyType(SingleStateStrategy.class)
    void showError();

    @StateStrategyType(SingleStateStrategy.class)
    void showEmptyView();

    @StateStrategyType(OneExecutionStateStrategy.class)
    void openDetailsScreen(int id);

    @StateStrategyType(AddToEndStrategy.class)
    void updateCertainNewsItemInList(@NonNull DbNewsItem item, int position);

    @StateStrategyType(AddToEndStrategy.class)
    void deleteNewsItemInList(int position);
}

File: app/src/main/java/com/androidacademy/msk/exerciseproject/screen/newslist/NewsListView.kt
package com.androidacademy.msk.exerciseproject.screen.newslist

import com.androidacademy.msk.exerciseproject.data.database.entity.DbNewsItem
import com.arellomobile.mvp.MvpView
import com.arellomobile.mvp.viewstate.strategy.OneExecutionStateStrategy
import com.arellomobile.mvp.viewstate.strategy.SingleStateStrategy
import com.arellomobile.mvp.viewstate.strategy.StateStrategyType

interface NewsListView : MvpView {
    @StateStrategyType(SingleStateStrategy::class)
    fun showProgressBar()

    @StateStrategyType(SingleStateStrategy::class)
    fun showNews(news: List<DbNewsItem>)

    @StateStrategyType(SingleStateStrategy::class)
    fun showError()

    @StateStrategyType(OneExecutionStateStrategy::class)
    fun openDetailsScreen(id: Int)

    @StateStrategyType(OneExecutionStateStrategy::class)
    fun setCurrentSectionInSpinner(position: Int)

    @StateStrategyType(OneExecutionStateStrategy::class)
    fun toggleSwipeRefreshProgress(show: Boolean)
}